exporter/containerimage/export.go

```diff
@@ -192,20 +192,28 @@ func (e *imageExporterInstance) Config() *exporter.Config {
 	return exporter.NewConfigWithCompression(e.opts.RefCfg.Compression)
 }
 
-func (e *imageExporterInstance) Export(ctx context.Context, src *exporter.Source, sessionID string) (_ map[string]string, descref exporter.DescriptorReference, err error) {
-	if src.Metadata == nil {
-		src.Metadata = make(map[string][]byte)
+func (e *imageExporterInstance) Export(ctx context.Context, src *exporter.Source, sessionID string) (map[string]string, exporter.DescriptorReference, error) {
+	return e.ExportImage(ctx, src, exptypes.InlineCache{}, sessionID)
+}
+
+func (e *imageExporterInstance) ExportImage(ctx context.Context, inp *exporter.Source, inlineCache exptypes.InlineCache, sessionID string) (_ map[string]string, descref exporter.DescriptorReference, err error) {
+	meta := make(map[string][]byte)
+	for k, v := range inp.Metadata {
+		meta[k] = v
 	}
 	for k, v := range e.meta {
-		src.Metadata[k] = v
+		meta[k] = v
 	}
+	src := *inp
+	src.Metadata = meta
```

Comment:
If we're making a copy, we should overwrite `inp` with the copy so we don't accidentally use it later.

Reply:
Done. I also updated the inputs to be values instead of references to present a stronger cue that they're not supposed to relay values back to caller.